internal/filter/filterconfig/config.go

```diff
@@ -126,9 +126,11 @@ type MatchProperties struct {
 }
 
 var (
-	ErrMissingRequiredField    = errors.New(`at least one of "attributes", "libraries",  or "resources" field must be specified`)
-	ErrInvalidLogField         = errors.New("services, span_names, and span_kinds are not valid for log records")
-	ErrMissingRequiredLogField = errors.New(`at least one of "attributes", "libraries", "span_kinds", "resources", "log_bodies", "log_severity_texts" or "log_severity_number" field must be specified`)
+	ErrMissingRequiredSpanField   = errors.New(`at least one of "attributes", "libraries",  or "resources" field must be specified`)
+	ErrInvalidLogField            = errors.New("services, span_names, span_kinds and metric_names are not valid for log records")
+	ErrMissingRequiredLogField    = errors.New(`at least one of "attributes", "libraries", "span_kinds", "resources", "log_bodies", "log_severity_texts" or "log_severity_number" field must be specified`)
+	ErrMissingRequiredMetricField = errors.New(`at least one of "metric_names" or "resources" field must be specified`)
+	ErrInvalidMetricField         = errors.New("span_names, span_kinds, resources, log_bodies, log_severity_texts, log_severity_number, services, attributes and libraries are not valid for metrics")
```

Comment:
```suggestion
	ErrInvalidMetricField         = errors.New(`"span_names", "span_kinds", "log_bodies", "log_severity_texts", "log_severity_number", "services", "attributes" and "libraries" are not valid for metrics`)
```
From what I can tell based on the code and one of your comments, `resources` should be allowed.

Also we should try to quote these like the other error messages.

Reply:
nice catch, adapted